private/bufpkg/bufimage/bufimagemodify/java_outer_classname.go

```diff
@@ -65,11 +66,20 @@ func javaOuterClassnameForFile(
 	sweeper Sweeper,
 	imageFile bufimage.ImageFile,
 	javaOuterClassnameValue string,
+	preserveExistingValue bool,
 ) error {
+	if isWellKnownType(ctx, imageFile) {
+		// The file is a well-known type - don't override the value.
```

Comment:
What about if someone has the WKTs vendored? They might want to override in this case (I know this is pre-existing logic, but wanted to bring it up)

Reply:
Haven't thought about it but I'd guess most people would expect to resolve the files from protobuf-java/protobuf-javalite artifact, which uses whatever settings which are present in the files. In most cases they do use pascal case so overriding might be a no-op, but in a few cases they are pluralized which wouldn't match our algorithm:

```
google/protobuf/descriptor.proto
45:option java_outer_classname = "DescriptorProtos";
...
google/protobuf/compiler/plugin.proto
48:option java_outer_classname = "PluginProtos";
```

I think keeping this safeguard in place is probably a good thing.